crowbar_framework/app/models/service_object.rb

```diff
@@ -1725,4 +1741,38 @@ def active_update(proposal, inst, in_queue, bootstrap = false)
       [400, e2.message]
     end
   end
+
+  def skip_unready_nodes(bc, inst, new_elements, old_elements)
+    skip_unready_nodes_config = Rails.application.config.experimental["skip_unready_nodes"]
+    # return if the feature is disabled
+    return new_elements unless skip_unready_nodes_config["enabled"]
+    logger.warn("Experimental skip_unready_nodes feature enabled")
+    logger.debug("skip_unready_nodes: enter for #{bc}:#{inst}")
+    nodes_pre_cached = {}
+    cleaned_elements = new_elements.deep_dup
+    skip_unready_nodes_config["roles"].each do |role|
+      # only do something if we have the same role on both old and new
+      next unless new_elements.key?(role) || old_elements.key?(role)
+      # we only can skip nodes that are on both old and new, as we know that those old nodes had
+      # the roles applied and will eventually become consistent with the deployment due to the
+      # periodic chef run
+      shared_elements = new_elements[role] & old_elements[role]
+      shared_elements.each do |n|
+        node = Node.find_by_name(n)
```

Comment:
Please do something like

```ruby
nodes_pre_cached[n] ||= Node.find_by_name(n)
node = nodes_pre_cached[n]
```
This matters for nodes that exist in multiple roles in the same proposal.

Btw, any reason to have used `nodes_pre_cached` instead of `pre_cached_nodes` (which is used everywhere else)?

Reply:
ah yeah, good catch.

> Btw, any reason to have used nodes_pre_cached instead of pre_cached_nodes (which is used everywhere else)?

Im just that dumb and thougth `nodes_pre_cached` was the actual var, never got to check it back :)